ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -40,97 +40,171 @@ const styles = reactStyles({
+      incrementButtonClassName='p-button-secondary'
+      value={updatedDiskSize || diskSize}
+      inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+      onChange={({value}) => onChange(value)}
+      min={50 /* Runtime API has a minimum 50GB requirement. */}/>
+  </Fragment>;
+};
+
+const DataProcConfig = ({onChange, dataprocConfig}) => {
+  const {
+    workerMachineType = defaultMachineType,
+    workerDiskSize = 50,
+    numberOfWorkers = 2,
+    numberOfPreemptibleWorkers = 0
+  } = dataprocConfig || {};
+  const [updatedNumWorkers, setUpdatedNumWorkers] = useState(numberOfWorkers);
+  const [updatedPreemtible, setUpdatedPreemptible] = useState(numberOfPreemptibleWorkers);
+  const [updatedWorkerMachine, setUpdatedWorkerMachine] = useState(null);
+  const [updatedDiskSize, setUpdatedDiskSize] = useState(workerDiskSize);
+
+  useEffect(() => {
+    const machineType = updatedWorkerMachine && updatedWorkerMachine.name;
+    const dataprocConfigChanged = updatedNumWorkers !== numberOfWorkers ||
+    updatedPreemtible !== numberOfPreemptibleWorkers ||
+    updatedDiskSize !== workerDiskSize ||
+    updatedWorkerMachine ||
+    !dataprocConfig;
+
+    onChange(dataprocConfigChanged ? {
+      workerMachineType: machineType,
+      workerDiskSize: updatedDiskSize,
+      numberOfWorkers: updatedNumWorkers,
+      numberOfPreemptibleWorkers: updatedPreemtible
+    } : null);
+
+    return () => onChange(null);
+  }, [updatedNumWorkers, updatedPreemtible, updatedWorkerMachine, updatedDiskSize]);
+
+
+  return <fieldset style={{marginTop: '0.75rem'}}>
+    <legend style={styles.workerConfigLabel}>Worker Config</legend>
+    <div style={styles.formGrid}>
+      <label htmlFor='num-workers' style={{marginRight: '.25rem'}}>Workers</label>
+      <InputNumber id='num-workers'
+        showButtons
+        decrementButtonClassName='p-button-secondary'
+        incrementButtonClassName='p-button-secondary'
+        value={updatedNumWorkers}
+        inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+        onChange={({value}) => {
+          setUpdatedNumWorkers(value);
+          if (updatedNumWorkers < updatedPreemtible) {
+            setUpdatedPreemptible(updatedNumWorkers);
+          }
+        }}
+        min={2}/>
+      <label htmlFor='num-preemptible' style={{marginRight: '.25rem'}}>Preemptible</label>
+      <InputNumber id='num-preemptible'
+        showButtons
+        decrementButtonClassName='p-button-secondary'
+        incrementButtonClassName='p-button-secondary'
+        value={updatedNumWorkers < updatedPreemtible ? updatedNumWorkers : updatedPreemtible}
+        inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+        onChange={({value}) => setUpdatedPreemptible(value)}
+        min={0}
+        max={updatedNumWorkers}/>
+      <div style={{gridColumnEnd: 'span 2'}}/>
+      <MachineSelector machineType={workerMachineType} onChange={setUpdatedWorkerMachine} updatedMachine={updatedWorkerMachine}/>
```

Comment:
Reusing this for the worker nodes is nice. This brings up an interesting point, as to whether we need to be as strict about machine types for different scenarios., The case they were definitely worried about with Terra UI was spark master nodes. However, the same hasn't been validated for GCE or worker nodes.

Could you add a TODO which references this ticket (which I just filed) here? https://precisionmedicineinitiative.atlassian.net/browse/RW-5763